packages/core/src/actions/actions.ts

```diff
@@ -56,6 +56,66 @@ export const UPDATE_I18N = 'jsonforms/UPDATE_I18N' as const;
 export const ADD_DEFAULT_DATA = 'jsonforms/ADD_DEFAULT_DATA' as const;
 export const REMOVE_DEFAULT_DATA = 'jsonforms/REMOVE_DEFAULT_DATA' as const;
 
+export type UpdateArrayContext =
+  | { type: 'ADD'; values: any[] }
+  | { type: 'REMOVE'; indeces: number[] }
+  | { type: 'MOVE'; moves: { from: number; to: number }[] };
+
+export const isUpdateArrayContext = (
+  context: object
+): context is UpdateArrayContext => {
+  if (!('type' in context)) {
+    return false;
+  }
+  if (typeof context.type !== 'string') {
+    return false;
+  }
+  switch (context.type) {
+    case 'ADD': {
+      if (
+        'values' in context &&
+        Array.isArray(context.values) &&
+        context.values.length > 0
+      ) {
+        const firstValueType = typeof context.values[0];
+        return context.values.every((v) => typeof v === firstValueType);
+      }
+      return false;
+    }
+    case 'REMOVE': {
+      if (
+        'indeces' in context &&
+        Array.isArray(context.indeces) &&
+        context.indeces.length > 0 &&
+        context.indeces.every((i) => typeof i === 'number')
+      ) {
+        return true;
+      }
+      return false;
```

Comment:
We could just return the if-condition instead of returning true and false explicitly ;)